### DISCUSSION TITLE
app server understanding-problem

### DISCUSSION THREAD

**Discussion by RuppAutomation** [bacnet-stack/bacnet-stack#883] [Q&A]
The server app creates dynamical following objects:
Max APDU: 1476
Created object analog-input-1
Created object analog-output-1
Created object analog-value-1
Created object binary-input-1
Created object binary-output-1
Created object binary-value-1
Created object calendar-1
Created object multi-state-input-1
Created object multi-state-output-1
Created object multi-state-value-1
Created object life-safety-point-1
Created object life-safety-zone-1
Created object load-control-1
Created object structured-view-1
Created object bitstring-value-1
Created object channel-1
Created object lighting-output-1
Created object binary-lighting-output-1

this is clear for me.

but why there will be create many more

what i have todo to solve these behaviour?
![Unbenannt](https://github.com/user-attachments/assets/70071ae4-ce39-4f8a-95d0-ac76b2399a99)

**Comment by RuppAutomation** [comment]
forget it, i got my answer by checking the source.

Is it right that, lets say for example NC is completly to implement by my self?

**Comment by skarg** [comment]
The example apps/server utilizes two methods to simulate objects using the basic objects in the source tree.

1. CreateObject service - for those basic objects that support CreateObject service, one of each object is created
2. For those basic objects that don't support CreateObject service, each object has a 'define' that the compiler uses to create a fixed number of objects.

For deeply embedded and products that forbid malloc(), CreateObject service is not always welcome.  However, I don't find any difference in creating an object that will never be deleted, and having a fixed number of objects.  It is easier to maintain and reuse basic objects that support CreateObject service. As time permits, all the basic example objects will eventually support CreateObject service.

For most BACnet products I have developed, I have copied the basic objects and customized them for the product and customers.  If the basic object - as implemented in the source tree - enables you to model your product for your customer, then you won't have to change anything for that object. However, there are a lot of optional properties and optional features and interdependencies on those features that are not implemented in the basic objects in this library.  If your product requires those additional features you can either implement them yourself, or you can use a more full featured commercial library.
